examples/sampling/pyg/node_classification.py

```diff
@@ -0,0 +1,210 @@
+
+    item_sampler = gb.ItemSampler(dataset_set, batch_size=1024, shuffle=shuffle)
+    neighbor_sampler = item_sampler.sample_neighbor(graph, [10, 10, 10])
+    feature_fetcher = gb.FeatureFetcher(
+        neighbor_sampler, feature, node_feature_keys=["feat"]
+    )
+    datapipe = gb.CopyTo(feature_fetcher, device=device)
+    data_loader = gb.DataLoader(datapipe, num_workers=0)
+    return data_loader
+
+
+def compute_accuracy(logits, labels):
+    _, predicted = torch.max(logits, dim=1)
+    correct = (predicted == labels).sum().item()
+    return correct / labels.size(0)
+
+
+def train(model, dataloader, optimizer, criterion, device):
+    #####################################################################
+    # (HIGHLIGHT) Train the model for one epoch.
+    #
+    # - Iterates over the data loader, fetching mini-batches of graph data.
+    # - For each mini-batch, it performs a forward pass, computes loss, and
+    #   updates the model parameters.
+    # - The function returns the average loss and accuracy for the epoch.
+    #
+    # Parameters:
+    #   model: The GraphSAGE model.
+    #   dataloader: DataLoader that provides mini-batches of graph data.
+    #   optimizer: Optimizer used for updating model parameters.
+    #   criterion: Loss function used for training.
+    #   device: The device (CPU/GPU) to run the training on.
+    #####################################################################
+
+    model.train()  # Set the model to training mode
+    total_loss = 0  # Accumulator for the total loss
+    total_correct = 0  # Accumulator for the total number of correct predictions
+    total_samples = 0  # Accumulator for the total number of samples processed
+    num_batches = 0  # Counter for the number of mini-batches processed
+
+    for minibatch in dataloader:
+        node_features = minibatch.node_features["feat"]
+        labels = minibatch.labels
+        optimizer.zero_grad()
+        out = model(minibatch.blocks, node_features, device)
+        loss = criterion(out, labels)
+        total_loss += loss.item()
+        total_correct += compute_accuracy(out, labels) * labels.size(0)
+        total_samples += labels.size(0)
+        loss.backward()
+        optimizer.step()
+        num_batches += 1
+    avg_loss = total_loss / num_batches
+    avg_accuracy = total_correct / total_samples
+    return avg_loss, avg_accuracy
+
+
+@torch.no_grad()
+def evaluate(model, dataloader, device):
+    model.eval()
+    total_correct = 0
+    total_samples = 0
+    with torch.no_grad():
+        for minibatch in dataloader:
+            node_features = minibatch.node_features["feat"]
+            labels = minibatch.labels
+            out = model(minibatch.blocks, node_features, device)
+            total_correct += compute_accuracy(out, labels) * labels.size(0)
+            total_samples += labels.size(0)
+    return total_correct / total_samples
```

Comment:
can we utilize https://github.com/dmlc/dgl/blob/b9cf36c3753ba4852be80348addb233bb70915bd/examples/sampling/graphbolt/node_classification.py#L267 instead of computing manually?

Reply:
for sure